blog/2024-03-28-hybrid-search-typescript/index.mdx

```diff
@@ -0,0 +1,305 @@
+image: ./img/hero.png
+description: "Hybrid Search for curious Web Developers with the new Weaviate Typescript client and Next.js"
+---
+
+![A Web Developers Guide to Hybrid Search Cover](./img/hero.png)
+
+:::info 
+
+This tutorial is built with Next.js and Weaviate. It uses a beta version of the v3 Typescript Client. Find out more about the new client here - [Experimental clients](/developers/weaviate/client-libraries/typescript/typescript-v3.md).
+
+:::
+
+
+When the question of search arises, two techniques tend to take up a lot of the conversations. You would most likely ask if you need Keyword Search or Vector Search. Sometimes, the answer is both, which luckily for us happens to be Hybrid Search; the best of both worlds! 
+In this article, we’ll get into the technology that powers Hybrid Search, learn how to use it Weaviate, and build a Next.js Application powered by Hybrid Search in Weaviate. 
+
+## Getting into Hybrid Search? 
+
+To understand Hybrid Search, we need to understand the two technologies that power it. Vector Search and Keyword Search.
+
+### Vector Search
+
+Vector search, also known as Semantic Search, uses machine learning to grasp text context, not just keywords. It converts text to numerical vectors, finding matches based on conceptual similarity for accurate, relevant search results.
+
+![Semantic Search Vector Space](./img/semantic-search.png)
+
+
+### Keyword Search
+
+
+Keyword search finds objects with an exact match between the terms in the query and the documents. Weaviate uses the BM25/F function to calculate the relevancy of a document to a query, which uses both the term frequency and inverse document frequency.
+
+![Keyword Search Illustrated](./img/keyword-search.png)
+
+
+### The Best of Both Worlds
+
+
+Hybrid search in Weaviate combines keyword (BM25) and vector search to leverage both exact term matching and semantic context. With either rankedFusion or relativeScoreFusion algorithms, we merge query results. Merging results in this way helps us have a better balance between precision querying in keyword search and the semantic understanding of Vector Search. 
+
+A code snippet of Hybrid Search in Javascript will look something like this. 
+
+```typescript
+const myCollection = client.collections.get('JeopardyQuestion');
+
+const result = await myCollection.query.hybrid('food',{
+ limit: 2,
+ returnProperties: ['question', 'answer'],
+})
+
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+
+## Optimizing your Hybrid Search
+
+With Weaviates Hybrid Search, we have two ways to adjust the Hybrid Search algorithm. We can adjust the weights by tweaking our search alpha or changing our ranking algorithm.
+
+### Changing alpha
+
+Think of alpha as a measure of how much your hybrid search should lean towards either Vector Search or keyword search where an alpha of 1 is a pure vector search, and an alpha of 0 is a pure keyword search. It is 0.75 by default. 
+
+
```

Comment:
Can we add a code snippet here using the alpha param in Typescript?

Reply:
yess!